test/integration/policy_selector_test.go

```diff
@@ -0,0 +1,230 @@
+					"value": `+string(selector)+`}]`)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			It(policyName+" should have the expected status", func(ctx SpecContext) {
+				Eventually(common.GetLatestStatusMessage(policyName, 0),
+					defaultTimeoutSeconds, 1).Should(Equal(test.status))
+			})
+		}
+
+		///////////////////
+		// Run the tests //
+		///////////////////
+
+		BeforeAll(func(ctx SpecContext) {
+			By("The ConfigMaps should be created on the Managed cluster")
+			By(fmt.Sprintf("Create namespace %s on the Managed cluster", configNamespace))
+			namespace := &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{Name: configNamespace},
+			}
+
+			_, err := clientManaged.CoreV1().Namespaces().Create(ctx, namespace, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			for i, name := range configMapNames {
+				By(fmt.Sprintf("Creating ConfigMap %s/%s on the Managed cluster", configNamespace, name))
+				configMap := &corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:   name,
+						Labels: map[string]string{selectorKey: name},
+					},
+					Data: map[string]string{"name": fmt.Sprintf("%s%d", "testvalue", i)},
+				}
+
+				_, err := clientManaged.CoreV1().ConfigMaps(configNamespace).Create(
+					ctx, configMap, metav1.CreateOptions{})
+				if !k8serrors.IsAlreadyExists(err) {
+					Expect(err).ToNot(HaveOccurred())
+				}
+
+				By(fmt.Sprintf("Creating ConfigMap %s/%s on the Managed cluster", configNamespace, unrelatedConfigMap))
+				_, err = clientManaged.CoreV1().ConfigMaps(configNamespace).Create(ctx, &corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: unrelatedConfigMap,
+					},
+				}, metav1.CreateOptions{})
+				if !k8serrors.IsAlreadyExists(err) {
+					Expect(err).ToNot(HaveOccurred())
+				}
+			}
+		})
+
+		DescribeTableSubtree("Reporting the correct status", baseSelectorTest, testEntries)
+
+		Describe("Reporting correct status with object updates", func() {
+			// Run a predefined base set of checks, including applying the policy
+			baseSelectorTest(defaultTest)
+
+			// Test definition for adding/removing matching ConfigMaps
+			cmChangeTest := func(crud string) {
+				It("when a matching ConfigMap is "+crud+"d", func() {
+					extraConfigMap := fmt.Sprintf("%s%d-extra", configMapName, len(configMapNames))
+					_, err := common.OcManaged(crud, "--namespace", configNamespace, "configmap", extraConfigMap)
+					Expect(err).ToNot(HaveOccurred())
+
+					configmaps := configMapNames
+					if crud == "create" {
+						configmaps = append(configmaps, extraConfigMap)
+					}
+
+					Eventually(common.GetLatestStatusMessage(policyName, 0),
+						defaultTimeoutSeconds, 1).Should(Equal(fmt.Sprintf("NonCompliant; violation - configmaps "+
+						"[%s] "+
+						"found but not as specified in namespace policy-selector-test",
+						strings.Join(configmaps, ", "))))
+				})
+			}
+
+			// Test creating and deleting ConfigMaps
+			cmChangeTest("create")
+			cmChangeTest("delete")
```

Comment:
Will we need more tests for other CRUD operations? If not, I feel like the function can just be expanded for these two cases and it would be simpler to read.

Reply:
Yeah, there probably won't be other CRUD operations (since updates are pretty well tested elsewhere).